frontend/src/pages/Scan/Scan.tsx

```diff
@@ -0,0 +1,245 @@
+import React, { useEffect, useState, useCallback } from 'react';
+import { useParams } from 'react-router-dom';
+import { useAuthContext } from 'context';
+import classes from './styles.module.scss';
+import { Organization as OrganizationType, Scan, ScanSchema } from 'types';
+import {
+  Header,
+  Button,
+  Form,
+  Dropdown,
+  TextInput,
+  Checkbox,
+  Label
+} from '@trussworks/react-uswds';
+import MultiSelect from '../Scans/MultiSelect';
+import { OrganizationOption } from 'pages/Scans/ScansView';
+import { Link } from 'react-router-dom';
+
+interface Errors extends Partial<OrganizationType> {
+  global?: string;
+}
+
+const ScanComponent: React.FC = () => {
+  const { scanId } = useParams();
+  const { apiGet, apiPut } = useAuthContext();
+  const [scan, setScan] = useState<Scan>();
+  const [errors, setErrors] = useState<Errors>({});
+  const [message, setMessage] = useState<string>('');
+  const [organizationOptions, setOrganizationOptions] = useState<
+    OrganizationOption[]
+  >([]);
+  const [scanSchema, setScanSchema] = useState<ScanSchema>({});
+  const [values, setValues] = useState<{
+    name: string;
+    organizations: OrganizationOption[];
+    arguments: string;
+    frequency: number;
+    frequencyUnit: string;
+    isGranular: boolean;
+    isSingleScan: boolean;
+  }>({
+    name: 'scan.name',
+    arguments: '{}',
+    organizations: [],
+    frequency: 1,
+    frequencyUnit: 'minute',
+    isGranular: false,
+    isSingleScan: false
+  });
+
+  const fetchScan = useCallback(async () => {
+    try {
+      const { scan, schema, organizations } = await apiGet<{
+        scan: Scan;
+        schema: ScanSchema;
+        organizations: OrganizationType[];
+      }>(`/scans/${scanId}`);
+      setScan(scan);
+      setScanSchema(schema);
+      setDefaultValues(scan);
+      setOrganizationOptions(
+        organizations.map((e) => ({ label: e.name, value: e.id }))
+      );
+    } catch (e) {
+      console.error(e);
+    }
+  }, [apiGet, setScan, scanId]);
+
+  const onSubmit: React.FormEventHandler = async (e) => {
+    e.preventDefault();
+    try {
+      let body: typeof values = Object.assign({}, values);
+      body.arguments = JSON.parse(values.arguments);
+      if (values.isSingleScan) body.frequency = 1;
```

Comment:
Can we refactor out the common logic in this file and the form in `ScansView.tsx`, so we're not repeating the same logic twice?

We could either refactor it out, or if you think it's easier, we could make "create a new scan" go to the "edit scan" page rather than having to have the scan form on the ScansView page.

Reply:
Put the logic in one place in Scan.tsx and imported it in ScansView